src/Command/BaseCommand.php

```diff
@@ -25,6 +26,16 @@ class BaseCommand extends Command
     const COMMAND_SUCCESS = 1;
     const COMMAND_FAILURE = 0;
 
+    public function run(InputInterface $input, OutputInterface $output)
+    {
+        $this->getApplication()->getDispatcher()->dispatch(
+            GushEvents::DECORATE_DEFINITION,
+            new ConsoleCommandEvent($this, $input, $output)
```

Comment:
ConsoleEvent is a base class for the ConsoleCommandEvent, ConsoleTerminateEvent and ConsoleExceptionEvent.